block_explosives remove remnants when detached.

Pickle-Coding opened the pull request:
Removes the explosive resistance remnants on the turf(s) if blocks_explosives gets detached.
## About The Pull Request
Closes #75793

Also this is an untested webedit.
## Why It's Good For The Game
hhhhhhhhhhhhhhhhh
## Changelog
:cl:
fix: Fix movable explosive blockers leaving remnants of explosive protection on turfs after they get destroyed.
/:cl:

Pickle-Coding committed:
block_explosives remove remnants when detached.

Removes the explosive resistance remnants on the turf(s) if blocks_explosives gets detached.

Pickle-Coding commented on code/datums/elements/blocks_explosives.dm:
```diff
@@ -20,6 +20,13 @@
+			for(var/atom/location as anything in target.locs)
+				unblock_loc(location, target.explosion_block)
+		else if(target.loc)
+			unblock_loc(location, target.explosion_block)
```
```suggestion
			unblock_loc(target.loc, target.explosion_block)
```

Pickle-Coding committed:
Update code/datums/elements/blocks_explosives.dm

Pickle-Coding commented on code/datums/elements/blocks_explosives.dm:
```diff
@@ -20,6 +20,13 @@
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
+		var/atom/movable/target = target
```
```suggestion
		var/atom/movable/target = source
```

Pickle-Coding replied:
Silly me.

Pickle-Coding committed:
Update code/datums/elements/blocks_explosives.dm

LemonInTheDark commented on code/datums/elements/blocks_explosives.dm:
```diff
@@ -20,6 +20,13 @@
 
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
+		var/atom/movable/target = source
+		if(length(target.locs) > 1)
+			for(var/atom/location as anything in target.locs)
+				unblock_loc(location, target.explosion_block)
+		else if(target.loc)
+			unblock_loc(target.loc, target.explosion_block)
```
won't work, detach is called when the element is explicitly removed.
this is still good to have, but to make this work with the issues you're looking at you'll need `ELEMENT_DETACH_ON_HOST_DESTROY` as an element flag

LemonInTheDark commented on code/datums/elements/blocks_explosives.dm:
```diff
@@ -20,6 +20,13 @@
 
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
```
this is unneeded, source cannot be anything but atom/movable here, so you can just drop this typecheck and the cast you do on the next line

Pickle-Coding replied:
Can I do the cast on the proc parameter without causing problems? I think the casting would be useful for code editors.

LemonInTheDark replied:
yep, that's fine byond won't care and only atoms should get there regardless

Pickle-Coding committed:
Call detach() when host destroy.

So it actually does the detach stuff when host gets deleted, for explosive resistance cleanup, lol.

Pickle-Coding committed:
Remove typecheck, cast in parameter instead.

Removes the typecheck in the Detach() proc, and typecasts in the parameter instead.
